File: problems/25-mst/001-min-cost-connect-points/java/Solution.java
import java.util.Arrays;

public class Solution {
    public int minCostConnectPoints(int[][] points) {
        int n = points.length;
        int[] minDistance = new int[n];
        boolean[] used = new boolean[n];
        Arrays.fill(minDistance, Integer.MAX_VALUE);
        minDistance[0] = 0;

        int total = 0;
        for (int count = 0; count < n; count++) {
            int current = -1;
            for (int i = 0; i < n; i++) {
                if (!used[i] && (current == -1 || minDistance[i] < minDistance[current])) {
                    current = i;
                }
            }

            used[current] = true;
            total += minDistance[current];

            for (int next = 0; next < n; next++) {
                if (!used[next]) {
                    int distance = Math.abs(points[current][0] - points[next][0])
                            + Math.abs(points[current][1] - points[next][1]);
                    minDistance[next] = Math.min(minDistance[next], distance);
                }
            }
        }

        return total;
    }
}

File: problems/25-mst/001-min-cost-connect-points/python/solution.py
class Solution:
    def minCostConnectPoints(self, points: list[list[int]]) -> int:
        n = len(points)
        min_distance = [float("inf")] * n
        used = [False] * n
        min_distance[0] = 0
        total = 0

        for _ in range(n):
            current = -1
            for i in range(n):
                if not used[i] and (current == -1 or min_distance[i] < min_distance[current]):
                    current = i

            used[current] = True
            total += min_distance[current]

            for next_node in range(n):
                if not used[next_node]:
                    distance = abs(points[current][0] - points[next_node][0]) + abs(
                        points[current][1] - points[next_node][1]
                    )
                    min_distance[next_node] = min(min_distance[next_node], distance)

        return total
